packages/core/src/invoke.ts

```diff
@@ -135,21 +122,21 @@ export function createActorRefFromInvokeAction<
       if (!IS_PRODUCTION) {
         warn(
           false,
-          `No behavior found for invocation '${src.type}' in machine '${machine.id}'.`
+          `Actor type '${src.type}' not found in machine '${machine.id}'.`
         );
       }
       return;
     }
 
     const behavior = behaviorCreator(context, _event.data, {
-      parent: parentRef as any, // TODO: fix
       id,
       data,
       src,
       _event
     });
 
     actorRef = new ObservableActorRef(behavior, id);
+    invokeAction.src = actorRef;
```

Comment:
Can you explain the mutation here?

Reply:
This was mentioned in the PR description as far from ideal - it was just something that I've hacked around to get me going and to get the final API shape (for this step of the refactor) passing tests.

I've refactored this now and this mutation doesn't happen anymore 🎉 Could you take another look? :)